test/performance/search/search_benchmark.cpp

```diff
@@ -0,0 +1,334 @@
+{
+    std::mt19937 gen{seed};
+    std::uniform_int_distribution<size_t> random_pos{0, std::ranges::size(seq) - overlap};
+    seq.erase(std::ranges::begin(seq) + random_pos(gen));
+}
+
+template <Alphabet alphabet_t>
+void mutate_substitution(std::vector<alphabet_t> & seq, size_t const overlap, size_t const seed)
+{
+    std::mt19937 gen{seed};
+    std::uniform_int_distribution<uint8_t> dis_alpha_short{0, alphabet_size<alphabet_t> - 2};
+    std::uniform_int_distribution<size_t> random_pos{0, std::ranges::size(seq) - overlap};
+    alphabet_t & cbase = seq[random_pos(gen)];
+    uint8_t crank = to_rank(cbase);
+    uint8_t rrank = dis_alpha_short(gen);
+    if (rrank >= crank)
+        ++rrank;
+    cbase.assign_rank(rrank);
+}
+
+template <Alphabet alphabet_t>
+auto generate_reads(std::vector<alphabet_t> const & ref,
+                    size_t const number_of_reads,
+                    size_t const read_length,
+                    uint8_t const simulated_errors_,
+                    double const prob_insertion,
+                    double const prob_deletion,
+                    double const standard_deviation = 0,
+                    size_t const seed = 0)
+{
+    std::vector<std::vector<alphabet_t>> reads;
+    std::mt19937 gen{seed};
+    std::uniform_int_distribution<size_t> seeds{0};
+    std::normal_distribution<> dis_error_count{static_cast<double> (simulated_errors_), standard_deviation};
+    std::uniform_real_distribution<double> probability(0.0, 1.0);
+
+    for (size_t i = 0; i < number_of_reads; ++i)
+    {
+        // TODO avoid mutating the same position multiple times
+        // simulate concrete error number or use normal distribution 
+        uint8_t simulated_errors = (standard_deviation == 0) ? simulated_errors_ : 
+                                                               abs(std::round(dis_error_count(gen)));
+        std::uniform_int_distribution<size_t> random_pos(0, std::ranges::size(ref) - read_length - simulated_errors);
+        size_t rpos = random_pos(gen);
+        std::vector<alphabet_t> read_tmp{std::ranges::begin(ref) + rpos,
+                                         std::ranges::begin(ref) + rpos + read_length + simulated_errors};
+
+        for (uint8_t j = 0; j < simulated_errors; ++j)
+        {
+            double prob = probability(gen);
+            // Substitution
+            if (prob_insertion + prob_deletion < prob)
+            {
+                mutate_substitution(read_tmp, simulated_errors, seeds(gen));
+            }
+            // Insertion
+            else if (prob_insertion < prob)
+            {
+                mutate_insertion(read_tmp, simulated_errors, seeds(gen));
+            }
+            // Deletion
+            else
+            {
+                mutate_deletion(read_tmp, simulated_errors, seeds(gen));
+            }
+        }
+        read_tmp.erase(std::ranges::begin(read_tmp) + read_length, std::ranges::end(read_tmp));
+        reads.push_back(read_tmp);
```

Comment:
Mhm, you always want the read to be 100 long?
But if the substitutions/insertions/deletion happen to be in `[100,100+simulated_errors)`, they wouldn't have any effect.

But it's probably ok since this will not happen to often and for a benchmark it's better to always have the same read length.

Reply:
By giving the mutate function an overlap parameter I make sure that mutations at pos >= 100 do not occure. But randomly inserting more insertions than deletions in the beginning still causes mutation at the very end of the read to be lost.